momento/update_ttl.go

```diff
@@ -0,0 +1,87 @@
+package momento
+
+import (
+	"context"
+	"time"
+
+	pb "github.com/momentohq/client-sdk-go/internal/protos"
+	"github.com/momentohq/client-sdk-go/responses"
+)
+
+type UpdateTtlRequest struct {
+	// Name of the cache to get the item from to be deleted
+	CacheName string
+	// string or byte key to be used to store item.
+	Key Key
+	// Time to live that you want to update in cache in seconds.
+	UpdateTtl time.Duration
+
+	grpcRequest  *pb.XUpdateTtlRequest
+	grpcResponse *pb.XUpdateTtlResponse
+	response     responses.UpdateTtlResponse
+}
+
+func (r *UpdateTtlRequest) cacheName() string { return r.CacheName }
+
+func (r *UpdateTtlRequest) key() Key { return r.Key }
+
+func (r *UpdateTtlRequest) updateTtl() time.Duration { return r.UpdateTtl }
+
+func (r *UpdateTtlRequest) requestName() string { return "UpdateTtl" }
+
+func (r *UpdateTtlRequest) initGrpcRequest(client scsDataClient) error {
+	var err error
+	var updateTtl uint64
+	defaultTtlMills := uint64(client.defaultTtl.Milliseconds())
+
+	var key []byte
+	if key, err = prepareKey(r); err != nil {
+		return err
+	}
+
+	if updateTtl, err = prepareUpdateTtl(r); err != nil {
+		return err
+	}
+	var request *pb.XUpdateTtlRequest
+	if updateTtl > defaultTtlMills {
+		request = &pb.XUpdateTtlRequest{CacheKey: key, UpdateTtl: &pb.XUpdateTtlRequest_IncreaseToMilliseconds{IncreaseToMilliseconds: updateTtl}}
+	} else if updateTtl < defaultTtlMills {
+		request = &pb.XUpdateTtlRequest{CacheKey: key, UpdateTtl: &pb.XUpdateTtlRequest_DecreaseToMilliseconds{DecreaseToMilliseconds: updateTtl}}
+	} else {
+		request = &pb.XUpdateTtlRequest{CacheKey: key, UpdateTtl: &pb.XUpdateTtlRequest_OverwriteToMilliseconds{OverwriteToMilliseconds: defaultTtlMills}}
+	}
```

Comment:
The SDK docs say "This API just sets the TTL for a key to the provided value.".

Does it work if we always overwrite?

```suggestion
	request = &pb.XUpdateTtlRequest{CacheKey: key, UpdateTtl: &pb.XUpdateTtlRequest_OverwriteToMilliseconds{OverwriteToMilliseconds: defaultTtlMills}}
```